addons/mail/models/mail_message.py

```diff
@@ -192,54 +192,29 @@ def _search_need_moderation(self, operator, operand):
     #------------------------------------------------------
 
     @api.model
-    def mark_all_as_read(self, channel_ids=None, domain=None):
-        """ Remove all needactions of the current partner. If channel_ids is
-            given, restrict to messages written in one of those channels. """
+    def mark_all_as_read(self, domain=None):
+        # not really efficient method: it does one db request for the
+        # search, and one for each message in the result set is_read to True in the
+        # current notifications from the relation.
         partner_id = self.env.user.partner_id.id
-        delete_mode = not self.env.user.share  # delete employee notifs, keep customer ones
-        if not domain and delete_mode:
```

Comment:
I just noticed that we don't have _log_access on mail_notification. So in order to do that we will need to add a field. I'll think a little about it, but if you want to try it:
Migration will need to be very fast, so this field must be empty by default. 
Write it on read; it will be a kind of read since. (not indexed)

Search for is_read=True, is_email=False, read_date < now - 6 months

I need to check if we have to filter them on partner_id.partner_share.

Reply:
@Xavier-Do 
can we use `mail_message_id.create_date` for searching notification instead of adding new field read_date in it??

Reply:
Yes we think about that but adding read_date has several advantages:
- Avoid a costly join. 
- The time before notification deletion may be shorter on our prod: Lets imagine a 1 months notification life, if you are on holidays or take some time to read it for other reason, the notification will disappear instantly. Removing it some time after read could be a functional improvement. 

We may change our mind later, since like you I feel that this field is not so useful, but it isn't so costly.
Note: we could do that in the mail.notification write so that we don't have to give date every time we write is_read.

Reply:
You can also add a config parameter to define the number of days before a notification is deleted (default to 60 for the moment.)

Reply:
@Xavier-Do 
https://github.com/odoo/odoo/pull/30067/commits/3170e7b268e82921b94e5c6d2f4eb9ac03ff80c8#diff-62124c1f25cf60267a7a5002dec5275cR54 we have pass param `days_before` in delete notification method, so it is easy to change from there rather than set it into the config param
what is your view on it?
Thanks

Reply:
Simple and good enough, works for me.